types/stylelint/index.d.ts

```diff
@@ -518,6 +518,7 @@ declare module 'stylelint' {
 						ruleSettings: ConfigRuleSettings<T, O>;
 						root: PostCSS.Root;
 						result?: PostcssResult;
+						context?: PluginContext;
```

Comment:
[suggest] It seems we've given an incorrect name. Perhaps `RuleContext` would be better than `PluginContext`.

Then, I'd like to mark `PluginContext` deprecated and add `RuleContext` as an alternative like this. What do you think?

```ts
export type RuleContext = {
	fix?: boolean | undefined;
	newline?: string | undefined;
};

/**
 * @deprecated Use `RuleContext` instead.
 */
export type PluginContext = RuleContext;
```

Reply:
Good idea, that is much more consistent with the other `Rule*` types:
https://github.com/stylelint/stylelint/blob/d91bb5b779ce6ca69058549d58195a9e82b7c8d1/types/stylelint/index.d.ts#L162-L200

Reply:
Updated here: https://github.com/stylelint/stylelint/pull/6466/commits/856b30f30224c30574660bffd9ceaafc111cfc4a

Is there a protocol for tracking/ensuring the `PluginContext` type is removed in the next major release?

Reply:
Thanks for improving the code.

> Is there a protocol for tracking/ensuring the `PluginContext` type is removed in the next major release?

No. I consider such a type as `PluginContext` private, but some plugins seem to use the type. So marking deprecated instead of removing seems helpful for plugin authors.

Honestly, I don't think we need to consider changes in internal types *breaking* because they're not a public API and affect only type-checking.

With the next major version v15.0.0 at least, the deprecated type will be removed.